Loop/Managers/DeviceDataManager.swift

```diff
@@ -1421,8 +1421,39 @@ extension DeviceDataManager {
                 }
             }
         }
-        // Wait up to 25 seconds for uploads triggered by these commands to finish
-        let _ = remoteDataServicesManager.waitForUploadsToFinish(timeout: .now() + TimeInterval(25))
+    }
+    
+    func activateRemoteOverride(_ remoteOverride: TemporaryScheduleOverride?) {
+        loopManager.mutateSettings { settings in settings.scheduleOverride = remoteOverride }
+        self.triggerBackgroundUpload(for: .overrides)
+    }
+    
+    func addRemoteCarbEntry(_ carbEntry: NewCarbEntry, completion: @escaping (_ result: CarbStoreResult<StoredCarbEntry>) -> Void) {
+        let backgroundTaskID = "Add Remote Carb Entry"
+        let backgroundTask = UIApplication.shared.beginBackgroundTask(withName: backgroundTaskID)
+        carbStore.addCarbEntry(carbEntry) { result in
+            self.triggerBackgroundUpload(for: .carb)
+            UIApplication.shared.endBackgroundTask(backgroundTask)
+            completion(result)
+        }
+    }
+    
+    func enactRemoteBolus(units: Double, completion: @escaping (_ error: Error?) -> Void = { _ in }) {
+        let backgroundTaskID = "Add Remote Bolus Entry"
+        let backgroundTask = UIApplication.shared.beginBackgroundTask(withName: backgroundTaskID)
+        self.enactBolus(units: units, activationType: .manualNoRecommendation) { error in
+            self.triggerBackgroundUpload(for: .dose)
+            UIApplication.shared.endBackgroundTask(backgroundTask)
+            completion(error)
+        }
+    }
+    
```

Comment:
A helper method to trigger an upload and attach a background task to it's completion.

Reply:
I was worried these background task IDs needed to be unique per invocation. But from what I can tell they are labels for the debugger.